src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -774,6 +799,34 @@ namespace winrt::TerminalApp::implementation
         focusedTab->ClosePane();
     }
 
+    // Method Description:
+    // - Close the terminal app with keys. If there are more
+    //   than one tab opened, show a warning dialog.
+    void TerminalPage::_CloseWindow()
+    {
+        if (_tabs.size() > 1)
+        {
+            _ShowCloseWarningDialog();
+        }
+        else
+        {
+            _CloseAllTabs();
+        }
+    }
+
+    // Method Description:
+    // - Remove all the tabs opened and the terminal will terminate
+    //   on its own when the last tab is closed
+    void TerminalPage::_CloseAllTabs()
+    {
+        int tabCount = static_cast<int>(_tabs.size());
```

Comment:
I believe you can accomplish this without using an integer for the sole purpose of when the loop is complete.

It is probably more sensible to use a reverse iterator over the existing `std::vector` of `_tabs` that holds the `std::shared_ptr<Tab>`s.

Reply:
_RemoveTabViewItem calculates the index of the current tab and erase it based on the index. Using reverse iterator make the code throws wired exception here. I wonder if you think this is a big concern ?

Reply:
Looks good to me. Thanks for the changes.